src/CloudNative.CloudEvents.NewtonsoftJson/JsonEventFormatter.cs

```diff
@@ -306,16 +344,43 @@ protected virtual void DecodeStructuredModeDataBase64Property(JToken dataBase64T
 
         public override byte[] EncodeStructuredModeMessage(CloudEvent cloudEvent, out ContentType contentType)
         {
-            cloudEvent = cloudEvent ?? throw new ArgumentNullException(nameof(cloudEvent));
-            cloudEvent.ValidateForConversion(nameof(cloudEvent));
-
             contentType = new ContentType("application/cloudevents+json")
             {
                 CharSet = Encoding.UTF8.WebName
             };
 
             var stream = new MemoryStream();
             var writer = new JsonTextWriter(new StreamWriter(stream));
+            WriteCloudEventForBatchOrStructuredMode(writer, cloudEvent);
+            writer.Flush();
+            return stream.ToArray();
+        }
+
+        public override byte[] EncodeBatchModeMessage(IEnumerable<CloudEvent> cloudEvents, out ContentType contentType)
+        {
+            cloudEvents = cloudEvents ?? throw new ArgumentNullException(nameof(cloudEvents));
+            contentType = new ContentType("application/cloudevents-batch+json")
```

Comment:
Consider putting this in a constants file along with the content type for structured mode.

Reply:
Definitely happy to make it a constant... not sure about a file just for constants though. I could put the prefix in MimeUtilities which is where most of our content type common code is. Will have a look.